### DISCUSSION TITLE
Is it the case that weight/capacity in min_cost_flow needs to be integer or with 1 decimal?

### DISCUSSION THREAD

**Discussion by cosmosanalytics** [networkx/networkx#5527] [Q&A] [answered]
The following simple problem can be solved when weights/capacities are integer
############################
import networkx as nx
G = nx.DiGraph()
G.add_node("3256", demand=-25)
G.add_node("0", demand=25)
G.add_edge("3256", "3795", weight=4015, capacity=6035)
G.add_edge("3256", "6565", weight=4885, capacity=1680)
G.add_edge("3795", "0")
G.add_edge("6565", "0")

flowDict = nx.min_cost_flow(G)
flowDict
############################

But when I change to float, it can't be solved??

############################
import networkx as nx
G = nx.DiGraph()
G.add_node("3256", demand=-25.0)
G.add_node("0", demand=25.0)
G.add_edge("3256", "3795", weight=4015.907896, capacity=6035.0)
G.add_edge("3256", "6565", weight=4885.247718, capacity=1680.0)
G.add_edge("3795", "0")
G.add_edge("6565", "0")

flowDict = nx.min_cost_flow(G)
flowDict
############################

yet the following problem can be solved ??
##########################
import networkx as nx
G = nx.DiGraph()
G.add_node("3256", demand=-25.0)
G.add_node("0", demand=25.0)
G.add_edge("3256", "3795", weight=4015.9, capacity=6035.0)
G.add_edge("3256", "6565", weight=4885.2, capacity=1680.0)
G.add_edge("3795", "0")
G.add_edge("6565", "0")

flowDict = nx.min_cost_flow(G)
flowDict
#############################

**Comment by rossbar** [comment, marked as answer]
This is a duplicate of #4641 - `min_cost_flow` uses `network_simplex` under the hood.

See also the [`Notes` section in the docstring](https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.flow.min_cost_flow.html) which acknowledges this problem and suggests alternatives to floating-point attributes.
